.github/workflows/build_and_push_image.yml

```diff
@@ -0,0 +1,83 @@
+name: Build and push narrative images
+
+on:
+  workflow_call:
+    secrets:
+      GHCR_USERNAME:
+        required: true
+      GHCR_TOKEN:
+        required: true
+
+jobs:
+  docker_build_image:
+    runs-on: ubuntu-latest
+    steps:
+    -
+      name: Check out GitHub Repo
+      if: github.event.pull_request.draft == false
+      with:
+        ref: "${{ github.event.pull_request.head.sha }}"
+      uses: actions/checkout@v2
+    -
+      name: Set up environment
+      run: |
+        if [ ${{ github.base_ref }} == 'master' ]; then
+            branch=''
+        else
+            branch="-""${{ github.base_ref }}"
+        fi
+        narrative_version=`grep '\"version\":' src/config.json.templ | awk '{print $2}' | sed 's/"//g'`
+        narrative_git_hash=`grep '\"git_commit_hash\":' src/config.json.templ | awk '{print $2}' | sed 's/"//g' | sed 's/,//'`
+        echo "APP_IMAGE_TAG=ghcr.io/${{ github.repository }}${branch}:pr-${{ github.event.pull_request.number }}" >> $GITHUB_ENV
+        echo "APP_VERSION_IMAGE_TAG=ghcr.io/${{ github.repository }}${branch}_version:pr-${{ github.event.pull_request.number }}" >> $GITHUB_ENV
+        echo "BRANCH=${branch}" >> $GITHUB_ENV
+        echo "BUILD_DATE=$(date -u +"%Y-%m-%dT%H:%M:%SZ")" >> $GITHUB_ENV
+        echo "NARRATIVE_GIT_HASH=${narrative_git_hash}" >> $GITHUB_ENV
+        echo "NARRATIVE_VERSION=${narrative_version}" >> $GITHUB_ENV
+        echo "PR=pr-${{ github.event.pull_request.number }}" >> $GITHUB_ENV
+        echo "VCS_REF=$(echo "${{ github.event.pull_request.head.sha }}" | cut -c -7)" >> $GITHUB_ENV
```

Comment:
Sets up a load of environment variables for subsequent steps of the workflow. `$GITHUB_ENV` is a GHA workflow-specific file that sets the environment.